backend/compact-connect/lambdas/provider-data-v1/handlers/providers.py

```diff
@@ -61,11 +68,16 @@ def query_providers(event: dict, context: LambdaContext):  # pylint: disable=unu
                     **config.data_client.get_providers_sorted_by_family_name(
                         compact=compact,
                         jurisdiction=jurisdiction,
+                        provider_name=provider_name,
                         scan_forward=scan_forward,
                         pagination=body.get('pagination')
                     )
                 }
             case 'dateOfUpdate':
+                if provider_name is not None:
+                    raise CCInvalidRequestException(
+                        'givenName and familyName are not supported for sorting by dateOfUpdate'
+                    )
```

Comment:
Why not?

Reply:
When we are accessing providers by the index that sorts by name, we can efficiently query for the segment of that index that contains the sort key values with the name fields we want. On the index that sorts by `dateOfUpdate` we lose that ability.